pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1242,43 +1207,69 @@ func (vs *VSphere) CreateVolume(volumeOptions *VolumeOptions) (volumePath string
 		return "", err
 	}
 
-	// vmdks will be created inside kubevols directory
-	kubeVolsPath := filepath.Clean(ds.Path(VolDir)) + "/"
-	err = makeDirectoryInDatastore(vs.client, dc, kubeVolsPath, false)
-	if err != nil && err != ErrFileAlreadyExist {
-		glog.Errorf("Cannot create dir %#v. err %s", kubeVolsPath, err)
-		return "", err
-	}
-	glog.V(4).Infof("Created dir with path as %+q", kubeVolsPath)
+	// Create a disk with the VSAN storage capabilities specified in the volumeOptions.StorageProfileData.
+	// This is achieved by following steps:
+	// 1. Create dummy VM if not already present.
+	// 2. Reconfigure the VM with the disk.
+        // 3. Detach the disk from the dummy VM.
```

Comment:
I dont think we need to detach the disk from the dummy VM. Its ok for the disk to be attached to the dummy VM as this will anyways be powered off. When there is a request to attach the volume to a k8s node, you can attach it to the new VM. Note that only the k8s node/VM will be powered on and will be reading/writing to the disk. The dummy VM is just a handle to the disks which is useful for deleting it. With this method, you can avoid some more VC API calls (example you dont have to detach the disk from dummyVM every time you create a disk)